tests/test_manager.py

```diff
@@ -0,0 +1,293 @@
+    """Tests for MultiAgentManager in async mode."""
+
+    MODE = "async"
+
+    echo_skill_id = PublicId("fetchai", "echo", "0.7.0")
+
+    def setup(self):
+        """Set test case."""
+        self.agent_name = "test_what_ever12"
+        self.working_dir = "MultiAgentManager_dir"
+        self.project_public_id = PublicId("fetchai", "my_first_aea", "0.11.0")
+        self.project_path = os.path.join(
+            self.working_dir, self.project_public_id.author, self.project_public_id.name
+        )
+        assert not os.path.exists(self.working_dir)
+        self.manager = MultiAgentManager(self.working_dir, mode=self.MODE)
+
+    def teardown(self):
+        """Tear down test case."""
+        self.manager.stop_manager()
+
+    def test_workdir_created_removed(self):
+        """Check work dit created removed on MultiAgentManager start and stop."""
+        assert not os.path.exists(self.working_dir)
+        self.manager.start_manager()
+        assert os.path.exists(self.working_dir)
+        self.manager.stop_manager()
+        assert not os.path.exists(self.working_dir)
+
+    def test_MultiAgentManager_is_running(self):
+        """Check MultiAgentManager is running property reflects state."""
+        assert not self.manager.is_running
+        self.manager.start_manager()
+        assert self.manager.is_running
+        self.manager.stop_manager()
+        assert not self.manager.is_running
+
+    def test_add_remove_project(self):
+        """Test add and remove project."""
+        self.manager.start_manager()
+
+        self.manager.add_project(self.project_public_id)
+
+        assert self.project_public_id in self.manager.list_projects()
+        assert os.path.exists(self.project_path)
+
+        with pytest.raises(ValueError, match=r".*was already added.*"):
+            self.manager.add_project(self.project_public_id)
+
+        self.manager.remove_project(self.project_public_id)
+        assert self.project_public_id not in self.manager.list_projects()
+
+        with pytest.raises(ValueError, match=r"is not present"):
+            self.manager.remove_project(self.project_public_id)
+
+        self.manager.add_project(self.project_public_id)
+        assert self.project_public_id in self.manager.list_projects()
+        assert os.path.exists(self.project_path)
+
+    def test_add_agent(self):
+        """Test add agent alias."""
+        self.manager.start_manager()
+
+        self.manager.add_project(self.project_public_id)
+
+        new_tick_interval = 0.2111
+
+        self.manager.add_agent(
+            self.project_public_id,
+            self.agent_name,
+            component_overrides=[
+                {
+                    "type": "skill",
+                    **self.echo_skill_id.json,
+                    "behaviours": {
+                        "echo": {
+                            "args": {"tick_interval": new_tick_interval},
+                            "class_name": "EchoBehaviour",
```

Comment:
`class_name` is not anymore a configurable field

Reply:
well on develop it still is; so let's wait until your PR is merged with changing this